pkg/apis/networking/v1alpha1/ingress_types.go

```diff
@@ -100,6 +100,10 @@ type IngressSpec struct {
 	// +optional
 	Rules []IngressRule `json:"rules,omitempty"`
 
+	// HTTPOption is the option of HTTP. It has the following two values:
+	// `HTTPOptionEnabled`, `HTTPOptionRedirected`
+	HTTPOption HTTPOption `json:"httpOption,omitempty"`
```

Comment:
Still feels like we should validate this *if* it is specified, we just can't require it.

Reply:
Adding a validation which validates it be `empty` or `HTTPOptionEnabled` or `HTTPOptionRedirected ` will do?

Reply:
Yeah, I think the validation before was probably mostly fine, we just had to allow empty since that's where things are today.

Reply:
Done